server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -467,10 +489,28 @@ public void uploadMetadata(
             );
             try {
                 IndexOutput indexOutput = storeDirectory.createOutput(metadataFilename, IOContext.DEFAULT);
+                Map<String, Version> segmentToLuceneVersion = new HashMap<>();
+                for (SegmentCommitInfo segmentCommitInfo : segmentInfosSnapshot) {
+                    SegmentInfo info = segmentCommitInfo.info;
+                    Set<String> segFiles = info.files();
+                    for (String file : segFiles) {
+                        segmentToLuceneVersion.put(file, info.getVersion());
+                    }
+                }
+
                 Map<String, String> uploadedSegments = new HashMap<>();
                 for (String file : segmentFiles) {
                     if (segmentsUploadedToRemoteStore.containsKey(file)) {
-                        uploadedSegments.put(file, segmentsUploadedToRemoteStore.get(file).toString());
+                        UploadedSegmentMetadata metadata = segmentsUploadedToRemoteStore.get(file);
```

Comment:
Can we check if the lucene version can be set as an instance variable in the initialisation of RemoteSegmentStoreDirectory class itself? This would depend on if the lucene version can change during lifecycle of this class anytime. If it does not, we can move the `writtenBy` field initialisation in the constructor itself.

Reply:
We can set the lucene version in the ctor of `RemoteSegmentStoreDirectory` - 
https://github.com/opensearch-project/OpenSearch/blob/68c1c86897a59fcffa77bfecc06641326e5e5199/server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java#L99-L109

Reply:
What will be the source of the Lucene version in this case?

Reply:
We set the current elasticsearch version and corresponding lucene version. AFAIK, for the new commits, this acts as the basis for the lucene version as well - https://github.com/opensearch-project/OpenSearch/blob/68c1c86897a59fcffa77bfecc06641326e5e5199/libs/core/src/main/java/org/opensearch/Version.java#L92-L94. Please check and see if this can simplify the code.

Reply:
Hey Ashish,
Resolving this based on our analysis where we observed that using `SegmentInfos` covers more cases. Please reopen if any changes are needed here.
Thanks!